jobs/export_process/worker.go

```diff
@@ -36,28 +36,43 @@ func MakeWorker(jobServer *jobs.JobServer, app AppIface) model.Worker {
 		}
 
 		outPath := *app.Config().ExportSettings.Directory
-		exportFilename := job.Id + "_export.zip"
+		id := model.NewId()
+		exportZipFilename := id + "_export.zip"
+		exportLogFilename := id + "_export.log"
+		zipReader, zipWriter := io.Pipe()
+		logReader, logWriter := io.Pipe()
 
-		rd, wr := io.Pipe()
+		errCh := make(chan *model.AppError, 2)
+		defer close(errCh)
 
-		errCh := make(chan *model.AppError, 1)
 		go func() {
-			defer close(errCh)
-			_, appErr := app.WriteFile(rd, filepath.Join(outPath, exportFilename))
+			_, appErr := app.WriteFile(zipReader, filepath.Join(outPath, exportZipFilename))
```

Comment:
How do we wait this goroutine? Since the defer function in `L:46` called right after `MakeWorker` function finished and if we can't finish `app.WriteFile`, we will try to send the error to a closed channel.

Reply:
But the `MakeWorker` function will not be finished until the `app.WriteFile` is finished. This is because when you look at `L:72`,  where we are waiting until we receive 2 values from the channel.